include/dd/ComplexTable.hpp

```diff
@@ -166,17 +175,20 @@ namespace dd {
             // search in (potentially) lower bucket
             if (lowerKey != key) {
                 ++lower_neighbors;
-                Entry* p_lower = find(table[lowerKey], val);
-                if (p_lower != nullptr) {
+                // buckets are sorted so we only have to look into the last entry of the lower bucket
+                Entry* p_lower = last_entry_table[lowerKey];
+                if (p_lower != nullptr && val - p_lower->value < TOLERANCE) {
```

Comment:
```suggestion
                if (p_lower != nullptr && std::abs(val - p_lower->value) < TOLERANCE) {
```
shouldn't the absolute value be used here? or is it really guaranteed that `val >= p_lower->value`?

Reply:
We can even guarentee `val > p_lower->value` :)